pkg/virt-controller/watch/snapshot/snapshot.go

```diff
@@ -185,6 +200,14 @@ func (ctrl *VMSnapshotController) updateVMSnapshotContent(content *snapshotv1.Vi
 	var volumeSnapshotStatus []snapshotv1.VolumeSnapshotStatus
 	var deletedSnapshots, skippedSnapshots []string
 
+	vmSnapshot, err := ctrl.getVMSnapshot(content)
+	if err != nil {
+		return 0, err
+	}
+	if vmSnapshotFailed(vmSnapshot) {
```

Comment:
Hmmm, kind of think that since the vm snapshot can never be restored (actually do we verify phase before restoring?) there is no point having volumesnapshots around.  They do take up resources on the storage backend

Reply:
@mhenriks I just updated the code with delete of the content when deadline exceeded and although if the snapshot has failed it wont get to "Ready" state I added a cause to the restore specifically also of failed phase. last 2 commits